packages/diagram-ui/src/features/update/updateModel.ts

```diff
@@ -109,6 +115,70 @@ export class UpdateModelCommand extends BaseUpdateModelCommand {
     ): CommandReturn {
         (newRoot as SRoot).changeRevision = (oldRoot as SRoot).changeRevision + 1;
         (newRoot as SRoot).sequenceNumber = (oldRoot as SRoot).sequenceNumber;
+        layoutViewportIfNecessary(oldRoot.index.all(), newRoot.index.all(), this.dispatcher);
         return super.performUpdate(oldRoot, newRoot, context);
     }
 }
+
+/**
+ * Fits the viewport to the new diagram size if too many changes occurred (i.e. the user pasted half a diagram into the editor).
+ *
+ * @param newElements the index storing all new model elements
+ * @param old the index storing all current model elements
+ * @param dispatcher the dispatcher to dispatch the 'fit to screen' action
+ */
+function layoutViewportIfNecessary(
+    old: FluentIterable<SModelElementImpl>,
+    newElements: FluentIterable<SModelElementImpl>,
+    dispatcher: IActionDispatcher
+) {
+    const updatedElements = new Set(newElements.map((update) => update.id));
+
+    // When we have so few updates, it isn't worth relayouting the viewport. We only want to lay it out if there are enough changes
+    if (updatedElements.size < 10) {
```

Comment:
not sure if I agree on this
for example, what if you paste a class diagram with a single class? then you would need to have autolayout
I think what we want if the distance (from difference) is less than 10, not if the size of the new diagram is less than 10
but maybe I'm wrong here

Reply:
(also the name updatedElements is at least for me a bit confusing, as this is all new elements, not just the updated ones, but I may miss your intention here)

Reply:
Okay, I removed the condition now.
Let's see if this condition would be useful or not.